sql/src/main/java/io/crate/analyze/SQLPrinter.java

```diff
@@ -113,25 +121,29 @@ public Void visitMultiSourceSelect(MultiSourceSelect mss, StringBuilder sb) {
         private Void printSelect(QueriedRelation relation, StringBuilder sb) {
             sb.append("SELECT ");
             addOutputs(relation, sb);
-            addFrom(sb, relation);
-            clauseAndQuery(sb, "WHERE", relation.where());
+            WhereClause whereClause = addFrom(sb, relation);
+            addQueryClauses(relation, whereClause, sb);
+            return null;
+        }
+
+        private void addQueryClauses(QueriedRelation relation, WhereClause whereClause, StringBuilder sb) {
+            // We need to convert any References back to keep any alias information
+            QuerySpec querySpec = relation.querySpec().copyAndReplace(
```

Comment:
Would it make sense to apply the transformation on the components individually so that we can keep referring to the properties on the relation? We want to get rid of the QuerySpec in its current form and these changes here will make that harder.